File: vanVlaanderenDerksen/QLJava/src/sc/ql/model/expressions/arithmetical/Multiply.java
package sc.ql.model.expressions.arithmetical;

import sc.ql.model.Expression;
import sc.ql.model.expressions.BinaryExpression;
import sc.ql.model.visitors.ExpressionVisitor;

public class Multiply extends BinaryExpression {

	public Multiply(Expression left, Expression right) {
		super(left, right);
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}

}

File: koopmans/lib/ql/ast/expression/binary_expression/arithmetic_expression/multiply.rb
module QL
  module AST
    class Multiply < ArithmeticExpression
      # def eval(left, right)
      #   IntegerLiteral.new(left * right)
      # end
      def accept(left, visitor)
        visitor.visit_multiply(left, self)
      end
    end
  end
end
